### DISCUSSION TITLE
Problem/error with Vietnamese datasets?

### DISCUSSION THREAD

**Discussion by goctoidongtien** [openai/whisper#961] [General]
Hello, I'm just started to use Whisper via [FreeSubtitles.ai](https://github.com/openai/whisper/discussions/548#discussion-4580711) and the result has some problem. 

A large part of the resulting subtitle file returned to me is not subtitle lines, but a call for viewers to subscribe to a YouTube channel. The line in the picture means "Please subscribe to the channel Ghien Mi Go so you don't miss interesting videos." Tried many different models but same result.

I assume this is an issue related to the data source imported into the model from the start. Many subtitles on Vietnamese YouTube videos are not actually subtitles. It's just dummy subtitles, with a single line calling viewers to subscribe to the youtube channel with timecode from start to finish. I think OpenAI should double check its multilingual data source, because it greatly affects the results. Hope it gets fixed soon.

![image](https://user-images.githubusercontent.com/125235812/218367925-61b363e5-2267-4dab-9f1e-499f990e9bae.png)

**Comment by mayeaux** [comment]
Sometimes certain models can go crazy with certain content, I would try another model and see if it makes the same errors

**Reply by DavidScann** [reply]
I've trial and error'd multiple models, and the "largest" one that seems to work fine without anomalies like these is base. The results that it gives though is extremely subpar and not at all suitable for actual usage.

**Comment by phineas-pta** [comment]
it isn't a problem specific to vietnamese, but rather a classic hallucination problem, see #679

**Reply by DavidScann** [reply]
Possibly. Though, this happens with any audio input, and this is the only output it gives (that ad line over and over again), so I think there's also a bit of a problem with the dataset. Have you tried out the proposed fix in the aforementioned issue?

**Reply by goctoidongtien** [reply]
Maybe but this problem seem like happen in less common languages, especially with Vietnamese because the data available is not so much.

**Reply by phineas-pta** [reply]
@DavidScann certainly it's a problem with dataset, unfortunately we can't do much about it

@goctoidongtien it happens to english, many people reported having "music in background" or many times "thanks for watching!" if you read more closely the thread i mentioned

**Reply by phineas-pta** [reply]
this is an example of combining different techniques to prevent hallucinations: https://github.com/EtienneAb3d/WhisperHallu

**Reply by procion1302** [reply]
Can confirm, I had the similar error in Indonesian with the line "Thanks for watching" in Indonesian.
The other errors were skipping a big part of transcription, as well as replacing some of them with weird repeating sounds like "Hee".
Smaller models have performed better (except the quality of transcribed parts was expectedly worse).

**Comment by ghost** [comment]


**Reply by phineas-pta** [reply]
what aspect u looking for ?

**Reply by ghost** [reply]
2 aspects:
1) Improve error rate from general purpose from off-the-shelf Whisper in Vietnamese language which will benefit Vietnamese community.
2) Finetune Whisper for medical domain specific using medical vocabulary database which will benefit medical group. 
=> Will be open source to benefit everyone

**Reply by phineas-pta** [reply]
seem like there'll be a lot of fine tuning, resulting in several models for each purpose, as fine tune multiple times gives bad accuracy

fine tuning is not very expensive (feasible with colab free tier) so i guess funding is more for data collecting & engineering, and possibly serving the models

**Reply by ghost** [reply]
IC, Thanks for your insights.
We possibly use funding for data engineering (AWS S3, DB, Serverless like Lambda, API Gateways, data crawling, etc..) and inferencing (EC2, Dockers, EKS, etc..) purposes as long as got the right direction for continuous improvement. 
Do you have any ideas why error rates are large with common dataset from Mozilla? 
We are focussing on boderline communities in Vietnam and Cambodia, so accents will come from south Vietnam and mix with Cambodia.

**Reply by phineas-pta** [reply]
to answer simply: unless u change model, to improve accuracy, u can train longer and throw more data into it

**Reply by phineas-pta** [reply]
the thing is nobody know what whisper training curve look like, hard to say whether it's under-/over-fitted

but with deep learning u should worry less about overfitting, more about data (quality + quantity)

small model maybe sufficient for single language yes, but u know how much more there's english speech vs vietnamese speech in training data ? 400k h vs 691 h ; well if u really have that much data for vietnamese better training from scratch instead of fine tune

**Reply by phineas-pta** [reply]
u have to request access to vlsp dataset

**Reply by phineas-pta** [reply]
wait u dont have any leadership/planning yet ?

i dont have any AI/ML real world experience yet, i can participate but i dont think im able to lead

**Reply by phineas-pta** [reply]
u can see email on my profile
